lib/3scale/backend/service.rb

```diff
@@ -277,7 +277,7 @@ def persist_attribute(attribute, value, ignore_nils = false)
       def persist_sets
         storage.sadd storage_key_by_provider(:ids), id
         storage.sadd encode_key("services_set"), id
-        storage.sadd encode_key("provider_keys_set"), provider_key
+        storage.sadd encode_key("provider_keys_set"), provider_key unless provider_key.nil?
```

Comment:
Why can provider key be nil? Should we conditionally add or should we raise an error?

Reply:
Replied at https://github.com/3scale/apisonator/pull/384#issuecomment-2109914809